test/cri-containerd/update_utilities.go

```diff
@@ -0,0 +1,93 @@
+// +build functional
+
+package cri_containerd
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/Microsoft/hcsshim/internal/shimdiag"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+const podJobObjectUtilPath = "C:\\jobobject-util.exe"
+
+func containerJobObjectName(id string) string {
+	return "\\Container_" + id
+}
+
+func createJobObjectsGetUtilArgs(ctx context.Context, cid, toolPath string, options []string) []string {
+	args := []string{"cmd", "/c", toolPath, "get"}
+	args = append(args, options...)
+	args = append(args, containerJobObjectName(cid))
+	return args
+}
+
+func checkLCOWResourceLimit(t *testing.T, ctx context.Context, client runtime.RuntimeServiceClient, cid, path string, expected uint64) {
+	cmd := []string{"cat", path}
+	containerExecReq := &runtime.ExecSyncRequest{
+		ContainerId: cid,
+		Cmd:         cmd,
+		Timeout:     20,
+	}
+	r := execSync(t, client, ctx, containerExecReq)
+	if r.ExitCode != 0 {
+		t.Fatalf("failed with exit code %d to cat path: %s", r.ExitCode, r.Stderr)
+	}
+	output := strings.TrimSpace(string(r.Stdout))
+	bytesActual, err := strconv.ParseUint(output, 10, 0)
+	if err != nil {
+		t.Fatalf("could not parse output %s: %s", output, err)
+	}
+	if bytesActual != expected {
+		t.Fatalf("expected to have a memory limit of %v, instead got %v", expected, bytesActual)
+	}
+}
+
+func checkWCOWResourceLimit(t *testing.T, ctx context.Context, runtimeHandler, shimName, cid, query string, expected uint64) {
+	shim, err := shimdiag.GetShim(shimName)
+	if err != nil {
+		t.Fatalf("failed to find shim %v: %v", shimName, err)
+	}
+	shimClient := shimdiag.NewShimDiagClient(shim)
+
+	// share the test utility in so we can query the job object
+	guestPath := podJobObjectUtilPath
+	if err := shareInUVM(ctx, shimClient, testJobObjectUtilFilePath, guestPath, false); err != nil {
```

Comment:
I'm confused a bit. If we've got a process isolated pod won't this fail? Shouldn't we be using the check below to skip this  (in addition to setting where we expect the binary to be like you're doing)?

Reply:
Good catch, fixing